sdk/storage/storage-file-datalake/src/clients.ts

```diff
@@ -922,7 +922,9 @@ export class DataLakePathClient extends StorageClient {
 
     const { span, spanOptions } = createSpan("DataLakePathClient-move", options.tracingOptions);
 
-    const renameSource = `/${this.fileSystemName}/${this.name}`; // TODO: Confirm number of /
+    // Be aware that decodeURIComponent("%27") = "'"; but encodeURIComponent("'") = "'".
+    // But since both ' and %27 work with the service here so we omit replace(/'/g, "%27").
+    const renameSource = `/${this.fileSystemName}/${encodeURIComponent(this.name)}`; // TODO: Confirm number of /
     const renameDestination = `/${destinationFileSystem}/${destinationPath}`; // TODO: Confirm encoding
```

Comment:
It would be worth to check the two TODOs here.

Reply:
Don't know what it is talking about. I think it just works now. Probably just to remove the TODOs. @XiaoningLiu as you are the author.

Reply:
Ok to remove the TODOs here.

Reply:
Removed.